contracts/cyberpunk/tests/integration.rs

```diff
@@ -53,6 +54,35 @@ fn execute_argon2() {
     assert!(gas_used < expected * 120 / 100, "Gas used: {}", gas_used);
 }
 
+// Test with
+// cargo integration-test debug_works -- --nocapture
+#[test]
+fn debug_works() {
+    let mut deps = mock_instance_with_gas_limit(WASM, 100_000_000_000_000);
+
+    let _res: Response =
+        instantiate(&mut deps, mock_env(), mock_info("admin", &[]), Empty {}).unwrap();
+
+    let msg = ExecuteMsg::Debug {};
+    let _res: Response = execute(&mut deps, mock_env(), mock_info("caller", &[]), msg).unwrap();
+
+    let start = SystemTime::now();
+    deps.set_debug_handler(move |msg, info| {
+        let gas = info.gas_remaining;
+        let runtime = SystemTime::now().duration_since(start).unwrap().as_micros();
+        eprintln!("{msg} (gas: {gas}, runtime: {runtime}µs)");
+    });
```

Comment:
Here I tried capturing a mutable reference like `&mut last_time` to get diffs between the calls. This would also be needed to open a file once and then write to it for every call. But to do that we need to change `Fn` to `FnMut` in `for<'a> Fn(/* msg */ &'a str, DebugInfo<'a>)` and then we can't clone the function anymore.

Since this whole Environment stuff changes with Wasmer 3, we can implement that later. Then it will probably be easier.

Reply:
Yeah was thinking about that too. I'll add a CHANGELOG entry and a test in cosmwasm-vm. But otherwise this is good.